extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AgroalVaultCredentialProviderPassword.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.agroal.runtime;
+
+import java.util.Properties;
+
+import io.agroal.api.security.SimplePassword;
+import io.quarkus.vault.CredentialProvider;
+
+public class AgroalVaultCredentialProviderPassword extends SimplePassword {
+
+    private CredentialProvider credentialProvider;
+
+    public AgroalVaultCredentialProviderPassword(String credentialProviderName, CredentialProvider credentialProvider) {
+        super(credentialProviderName);
+        this.credentialProvider = credentialProvider;
+    }
+
+    @Override
+    public Properties asProperties() {
+        return credentialProvider.getCredentials(getWord());
```

Comment:
See you want them plural too :).

Reply:
ah. I thought you were talking about the method.
credentials is more common according to https://english.stackexchange.com/a/252075 and https://www.vocabulary.com/dictionary/credential
so I renamed it to `CredentialsProvider`